src/main/java/duke/main/Storage.java

```diff
@@ -0,0 +1,89 @@
+package duke.main;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+public class Storage {
+    private final String storagePath;
+
+    public Storage(String path) {
+        this.storagePath = path;
+    }
+
+    /**
+     * Saves all task into backup file with path given in constructor.
+     *
+     * @param tasks The list of tasks to be saved.
+     * @throws IOException If creating directory and file throws error.
+     */
+    public void saveTasksToStorage(TaskList tasks) throws IOException {
+        if (Files.notExists(Paths.get(storagePath))) {
+            Files.createDirectories(Paths.get("data/"));
+            Files.createFile(Paths.get(storagePath));
+        }
+        FileWriter fw = new FileWriter(storagePath);
+        for (Task task: tasks) {
+            fw.write(task.convertToSaveFormat());
+            fw.write("\n");
+        }
+        fw.close();
+    }
+
+    /**
+     * Converts formatted string in backup file into a task instance..
+     *
+     * @param str Formatted string in backup file.
+     * @return A task instance.
+     */
+    public Task loadTask(String[] str) {
```

Comment:
The loadTask method and load method have similar meanings. It appears that the loadTask method is only called by the load method alone. Perhaps you could make the loadTask method private.